src/fheroes2/battle/battle_troop.cpp

```diff
@@ -763,6 +763,11 @@ uint32_t Battle::Unit::Resurrect( const uint32_t points, const bool allow_overfl
     return resurrect;
 }
 
+bool Battle::Unit::canShoot() const
+{
+    return isArchers() && !Modes( SP_BLIND | SP_PARALYZE | SP_STONE ) && !isHandFighting();
```

Comment:
There is a "unifying" constant:

```suggestion
    return isArchers() && !Modes( SP_BLIND | IS_PARALYZE_MAGIC ) && !isHandFighting();
```